lib/web_ui/dev/roll_fallback_fonts.dart

```diff
@@ -63,8 +63,12 @@ class RollFallbackFontsCommand extends Command<bool>
     final http.Client client = http.Client();
     final List<String> fallbackFonts = <String>[];
     final Map<String, Uri> urlForFamily = <String, Uri>{};
-    await _addApiFallbackFonts(client, fallbackFonts, urlForFamily);
-    await _addSplitFallbackFonts(client, fallbackFonts, urlForFamily);
+    fallbackFonts.addAll(
+      await _processSplitFallbackFonts(client, splitFallbackFonts, urlForFamily)
+    );
+    fallbackFonts.addAll(
+      await _processFallbackFonts(client, apiFallbackFonts, urlForFamily)
+    );
```

Comment:
Well, now you can revert most of the PR, and just do:
```suggestion
    await _addSplitFallbackFonts(client, fallbackFonts, urlForFamily);
    await _addApiFallbackFonts(client, fallbackFonts, urlForFamily);
```
🙂

Reply:
Yes, but... I dislike modifying an incoming array in a method (and using a global inside that method, it made the code harder to read! :P)

Reply:
I understand that and I feel the same. I've been thinking about ways to fix this (more below).

The problem is we now have a method that returns half the result (the family list), and modifies an incoming map to store the other half (`urlForFamily`). IMHO, this is worse.

One way to fix this is instead of returning a list family names, we should return a list of `Tuple(family, url)`. The rest of the code should use these tuples instead using a family list and a url map.

Reply:
Jeez, I was so horrified in the `fallbackFonts` array being mutated that I didn't notice the incoming map. Indeed, let me move this method to a tuple.

Reply:
I'm generally not a fan of building an object by mutation (e.g. it requires making the fields nullable and using `!` to access them later).

If we want to reduce duplication a bit, maybe make `_Font` a wrapper around `ProcessedFallbackFont`?
```dart
typedef _FontInfo = ({ String family, Uri uri });

class _Font {
  _Font(this.fontInfo, this.index, this.starts, this.ends);
  
  String get family => fontInfo.family;
  String get uri => fontInfo.uri;
 
  // ...
}
```

Reply:
The tree seems to be broken, so I have time to do this, let's go!

Reply:
Done!